[use cases] Index based use cases for core (#4797)
ElektraInitiative/libelektra: doc/usecases/core/UC_keyset_insert.md

  - [`KeySet` has been created](UC_keyset_create.md)
- **Main success scenario:**
  - Caller requests to insert a new `Key` with non-cascading name into `KeySet`
  - Namespace of `Key` is allowed by type of `KeySet`.
  - If necessary Core resizes `KeySet`
  - If necessary Core removes existing `Key` with same name
  - Core adds `Key` to `KeySet`
- **Alternative scenario:**
  - Caller requests to insert all `Key`s contained in one `KeySet` A into another `KeySet` B
  - Type of A matches type of B.
  - If necessary Core resizes B
  - If necessary Core removes all existing `Key`s contained in A and B from B
  - Core copies all `Key`s from A to B
- **Error scenario:**
  - Caller requests to insert a new `Key` with non-cascading name into `KeySet`

>   - Namespace of `Key` is prohibited by type of `KeySet`.

Reviewer: type of `KeySet` also needs to be in glossary, if we need it. Otherwise we could simply say:

```suggestion
  - Namespace of `Key` is prohibited in this `KeySet`.
```
